docs/nima/guides/webclient.adoc

```diff
@@ -0,0 +1,411 @@
+
+[source,bash]
+.Output:
+----
+{"message":"Hello David!"}
+----
+
+This time, a JsonObject is printed out in the console. It is possible to change the greeting itself by using a PUT
+request to `/greet/greeting` endpoint from the base URI.
+
+[source,java]
+.Modify the application greeting:
+----
+import javax.json.Json;
+
+JsonObject entity = Json.createObjectBuilder() // <1>
+    .add("greeting", "Bonjour")
+    .build();
+webClient.put()     // <2>
+    .path("/greet/greeting")
+    .submit(entity)     // <3>
+    .thenCompose(response -> webClient.get()    // <4>
+        .path("/greet/David")
+        .request(JsonObject.class))
+    .thenAccept(System.out::println)
+    .await();
+----
+<1> Create a JsonObject with key `greeting` and value `bonjour`.
+<2> Create a PUT request.
+<3> Submit the JsonObject created earlier.
+<4> Once done, make a GET call to verify the modification was processed to the greeting.
+
+According to the quickstart documentation, a JSON object can be sent to the application to change the greeting following
+this structure: `{"greeting" : "value"}`. The first three lines of code create the JsonObject with the required content.
+This time, we use the PUT request and submit methods to push the new greeting. One way to check the greeting modification
+is to execute GET request again and display obtained response. The `thenCompose` method will execute a GET request after
+the PUT request is executed.
+
+[source,bash]
+.Output:
+----
+{"message":"Bonjour David!"}
+----
+
+=== WebClient Metrics [[WebClient-Metrics]]
+
+WebClient, like other Helidon components, supports Metrics. The following example introduces the different metrics that
+can be used to measure WebClient activity. There are two ways to set up metrics: programmatically on the WebClient
+instance or manually using the configuration file.
+
+==== Add metrics dependency
+
+There is a specific dependency to use WebClient metrics in your application.
+
+[source,xml]
+.Add the following dependency to pom.xml:
+----
+<dependency>
+    <groupId>io.helidon.reactive.webclient</groupId>
+    <artifactId>helidon-reactive-webclient-metrics</artifactId>
+</dependency>
+----
+
+==== Set up metrics on WebClient instance
+
+It is possible to register metrics on WebClient directly into the code. The following example shows a general method
+that can be used with any metric.
+
+[source,java]
+.Example of metric creation:
+----
+import io.helidon.common.http.Http;
+import io.helidon.metrics.RegistryFactory;
+import io.helidon.reactive.webclient.metrics.WebClientMetrics;
+import io.helidon.reactive.webclient.spi.WebClientService;
+import org.eclipse.microprofile.metrics.MetricRegistry;
+import org.eclipse.microprofile.metrics|.Counter;
+                                       |.Meter;
+                                       |.Timer;
+                                       |.ConcurrentGauge;
```

Comment:
Here and below, can we convey that the developer can choose among different metrics types using legal Java syntax (maybe comment the succeeding lines or something?), so readers can copy and paste?